### ISSUE TITLE
Misleading interpolated percentiles for small sample sizes

### ISSUE THREAD

**Issue by TuggyNE** [taoensso/tufte#67]
Up until about 2.4.5 (I think), Tufte had simple behavior around percentiles with very low sample counts, such as 99%ile of a 5-sample pstats. It just reported the exact stat at the quantile closest to the desired percentile, which meant that several of the highest percentiles would be identical for small samples. (At the far end, of course, a 1-sample would have _all_ its percentiles the same.)

Current behavior seems to have changed, although I don't see it in the changelog. Now it appears to interpolate values. Concretely, the straightforward test
```clojure
(->> (dotimes [_ 2] (p :test (Thread/sleep (long (* 100 (rand))))))
     (tufte/profiled {}) peek deref)
```
produces results like this:
```
{:stats
 {:test
  {:min 45379080,
   :p25 5.298566525E7,
   :p50 6.05922505E7,
   :p75 6.819883575E7,
   :p90 7.27627869E7,
   :p95 7.428410395E7,
   :p99 7.550115759E7,
   :max 75805421,
   :n 2
   ,,,
  }
 }
 ,,,
}
```
Where, as you can see, everything but :max and :min are synthetic. For a 2-sample test, of course, nobody should expect to get good percentiles out. But if someone is running a test with 25 or 50 samples and sees a plausible p99 value, they may not realize that it doesn't actually exist: it's just a regression. Given the nature of timing percentiles, this is likely to fool them into thinking they have a much more robust system than they actually do.

As an example, take the mildly perverse fn here:
```clojure
(defn spiky []
  (Thread/sleep
   (condp >= (rand)
     0.01 100
     0.05 5
     0.10 4
     0.50 3
     0.99 2
     1)))
```
Two runs may give very different results but, importantly, neither run gives adequate warning that the behavior of the function is not well described by the stats:
```clojure
clj꞉user꞉> 
(->> (dotimes [_ 20] (p :test (spiky)))
     (tufte/profiled {}) peek tufte/format-pstats println)
pId           nCalls        Min      50% ≤      90% ≤      95% ≤      99% ≤        Max       Mean   MAD      Clock  Total

:test             20     2.06ms     3.09ms     4.45ms     4.52ms     4.53ms     4.53ms     3.22ms  ±24%    64.37ms    98%

Accounted                                                                                                  64.37ms    98%
Clock                                                                                                      65.63ms   100%

nil
clj꞉user꞉> 
(->> (dotimes [_ 20] (p :test (spiky)))
     (tufte/profiled {}) peek tufte/format-pstats println)
pId           nCalls        Min      50% ≤      90% ≤      95% ≤      99% ≤        Max       Mean   MAD      Clock  Total

:test             20     2.12ms     3.20ms     5.40ms    10.46ms    82.93ms   101.04ms     7.97ms ±117%   159.36ms    98%

Accounted                                                                                                 159.36ms    98%
Clock                                                                                                     161.93ms   100%

nil
```
(While the first run has no data that could possibly indicate a problem, it does have the false impression of filled-out data. The second run is worse, since it *does* have warning data, but the interpolation actively de-emphasizes that, discounting the genuine 99%ile stat by 20%.)

**Comment by ptaoussanis** [comment]
Hi @TuggyNE, thanks for the very clear report 🙏

This was an unintentional regression, will try get a fix + regression tests in early next week. In the meantime, can confirm that v2.4.5 should be unaffected - so should be good to use in the meantime.

Apologies for the trouble!

**Comment by TuggyNE** [comment]
Thanks for the update, @ptaoussanis! Glad this was able to help you fix it so quickly.

**Comment by ptaoussanis** [comment]
@TuggyNE This should be addressed in [v2.6.0](https://github.com/taoensso/tufte/releases/tag/v2.6.0), on Clojars now.

Again, sorry for the trouble!

Cheers :-)
